sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java

```diff
@@ -170,13 +181,40 @@
      * @throws HttpRequestException if {@code name}  name} or {@code version} is empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Secret>> getSecret(String name, String version) {
+    public Mono<Secret> getSecret(String name, String version) {
+        return withContext(context -> getSecretWithResponse(name, version))
+            .flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Get the specified secret with specified version from the key vault. The get operation is
+     * applicable to any secret stored in Azure Key Vault. This operation requires the {@code
+     * secrets/get} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Gets a specific version of the secret in the key vault. Subscribes to the call
+     * asynchronously and prints out the
+     * returned secret details when a response is received.</p>
+     * {@codesnippet com.azure.keyvault.secrets.secretclient.getSecretWithResponse#string-string}
+     *
+     * @param name The name of the secret, cannot be null
+     * @param version The version of the secret to retrieve. If this is an empty String or null, this
+     * call is equivalent to calling {@link #getSecret(String)}, with the latest version being
+     * retrieved.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value}
+     * contains the requested {@link Secret secret}.
+     * @throws ResourceNotFoundException when a secret with {@code name} and {@code version} doesn't
+     * exist in the key vault.
+     * @throws HttpRequestException if {@code name}  name} or {@code version} is empty string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Secret>> getSecretWithResponse(String name, String version) {
         return withContext(context -> getSecret(name, version, context));
     }
 
     Mono<Response<Secret>> getSecret(String name, String version, Context context) {
```

Comment:
Try to standardise on naming. This method should be called `getSecretWithResponse`.

Reply:
this is the internal method.
The public equivalent of this `getSecretWithResponse` is [here](https://github.com/Azure/azure-sdk-for-java/pull/4549/files#diff-dd6898a46e4ae1f96633760f24441baaR209)

Reply:
I understand - but it should still follow the naming convention.